pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,288 @@
+        self.write("VOLT DOWN")
+
+    # Current Settings --------------------------------------------------------
+
+    current = Instrument.control(
+        "CURR?",
+        "CURR %s",
+        "Output current in A. Range depends on instrument type.",
+    )
+
+    min_current = Instrument.measurement(
+        "CURR? MIN", "Minimum current in A."
+    )
+
+    max_current = Instrument.measurement(
+        "CURR? MAX", "Maximum current in A."
+    )
+
+    def current_to_min(self):
+        """Set current of the selected channel to its minimum value."""
+        self.write("CURR MIN")
+
+    def current_to_max(self):
+        """Set current of the selected channel to its maximum value."""
+        self.write("CURR MAX")
+
+    current_step = Instrument.control(
+        "CURR:STEP?", "CURR:STEP %s", "Current step in A."
+    )
+
+    def current_up(self):
+        """Increase current by one step."""
+        self.write("CURR UP")
+
+    def current_down(self):
+        """Decreases current by one step."""
+        self.write("CURR DOWN")
+
+    # Combined Voltage And Current Settings -----------------------------------
+
+    voltage_and_current = Instrument.control(
+        "APPL?",
+        "APPL %s, %s",
+        "Output voltage (V) and current (A).",
+    )
+
+    # Output Settings ---------------------------------------------------------
+
+    selected_channel_active = Instrument.control(
+        "OUTP:SEL?",
+        "OUTPUT:SEL %s",
+        "Set the selected channel to active or inactive or check its status.",
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    output_on = Instrument.control(
+        "OUTP:GEN?",
+        "OUTP:GEN %s",
+        "Set the output on or off or check the output status.",
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    # The following commands are for making it easier to change selected the
+    # selected channel and activate/deactivate them.
+
+    def activate_channel(self, channel):
+        """Activate a channel."""
+        # Save current selected channel before switching.
+        selected_channel = self.selected_channel
+        self.selected_channel = channel
+        self.selected_channel_active = True
+        # Restore previously selected channel.
+        self.selected_channel = selected_channel
+
+    def deactivate_channel(self, channel):
+        """Deactivate a channel."""
+        # Save current selected channel before switching.
+        selected_channel = self.selected_channel
+        self.selected_channel = channel
+        self.selected_channel_active = False
+        # Restore previously selected channel.
+        self.selected_channel = selected_channel
+
+    # Convenience functions to make turning on/off the output more natural.
+    def turn_output_on(self):
+        """Turn output on."""
+        self.output_on = True
+
+    def turn_output_off(self):
+        """Turn output off."""
+        self.output_on = False
```

Comment:
I have to disagree that these are "convenience functions". Also, in a pymeasure context, using properties is actually more "natural".
While we have methods in our instruments to do things that are difficult/complex with properties, we do not just mirror functionality that already exists in properties, as this is just confusing for the users. I'd rather you remove those.

Reply:
Removed the mirrored functionality.